scalameta/common/shared/src/main/scala/scala/meta/internal/trees/ast.scala

```diff
@@ -336,7 +350,54 @@ class AstNamerMacros(val c: Context) extends AstReflection with CommonNamerMacro
           }
           internal(...$internalArgss)
         }
-      """
+        """
+
+        // step 13a: generate additional binary compat Companion.apply
+        // generate new applies for each new field added
+        // with field A, B and additional binary compat ones C, D and E, we generate:
+        // apply(A, B, C), apply(A, B, C, D), apply(A, B, C, D, E)
+        val sortedBinaryCompatName = binaryCompatVars
+          .collect { case vr: ValDef => vr }
+          .sortBy { vr =>
```

Comment:
Performance nitpick: the lambda to `sortBy` gets evaluated many times per element. You can cache the results by doing a method like here https://github.com/twitter/bazel-multideps/blob/db19c92d13bce01bcd0189309564c79213995917/multideps/src/main/scala/multideps/diagnostics/MultidepsEnrichments.scala#L58

Reply:
The size of the list will always be small here so I suspect performance is not a concern. Just an FYI, feel free to ignore

Reply:
Currently it's mostly only a single element list, so it should not be an issue. But I haven't really thought about the performance here. 

I think I will leave as is, because I am afraid of really adding any more complexity here.